### PR TITLE
feat(inspector): add exclude filter on log search

### PR DESCRIPTION
**Pull request by Florent-Bouisset**
Let user filter the logs in inspector based on a text that should NOT appear in the log line.
Usefull for example if there is a library that use a lot console.log() and the log stream is polluted by irrelevant log. Filtering out this lines help reading the logs more easily.

image

### CODE REVIEW

**Comment by peaBerberian** [comment]
There is something that bothers me visually here is that it's not clear that regex/case modifiers do not act on the exclusion filter.
Maybe we might need to set it on a different line if a checkbox is checked or something?

What do you think?

**Comment by Florent-Bouisset** [comment]
> There is something that bothers me visually here is that it's not clear that regex/case modifiers do not act on the exclusion filter. Maybe we might need to set it on a different line if a checkbox is checked or something?
> 
> What do you think?

I have changed the styling so it's clearer that the case sensitive and regex button only apply to the first input:

image

**Comment by peaBerberian** [comment]
In your example, you probably wanted to write `XHR`, no?

Also, if it's written that way, why not also adding the exclude part its own regex/case buttons?

**Comment by Florent-Bouisset** [comment]
> In your example, you probably wanted to write `XHR`, no?

Yes 😅

> Also, if it's written that way, why not also adding the exclude part its own regex/case buttons?

Adding an "exclude" input doesn't enhance the filtering capabilities. The exclusion functionality was already achievable using the "filter" input with negative regex capturing groups, such as ^((?!foo).)*$. 

The "exclude" input doesn't introduce new filters that were previously unavailable, it just make things more easier
hiding some verbose logs.
That's why the inclusion of regex/case buttons wasn't deemed necessary.

**Comment by peaBerberian** [comment]
OK I get it. It may even be a first step of a whole group of advanced search filters (like filtering only network, warnings and so on).

In that I still found it unclear that buttons do not apply, as the text input is still positioned relative to it. Why not making it full width?

**Comment by Florent-Bouisset** [comment]
> OK I get it. It may even be a first step of a whole group of advanced search filters (like filtering only network, warnings and so on).

Yes we could think about that in the future

> In that I still found it unclear that buttons do not apply, as the text input is still positioned relative to it. Why not making it full width?

Yes you are right, I have changed it.

File: inspector/src/modules/log_module.ts [hunk 1]
```diff
@@ -117,15 +119,15 @@ export default function LogModule({
...
+  timeRangeInputElt.style.marginTop = "10px";
   timeRangeInputElt.style.overflow = "hidden";
   timeRangeInputElt.style.justifyContent = "space-between";
 
   /** Wrapper elements allowing to filter logs. */
   const filterFlexElt = strHtml`<div class="log-wrapper"/>`;
   filterFlexElt.style.display = "flex";
-  filterFlexElt.style.height = "40px";
   filterFlexElt.style.margin = "5px 0px";
+  filterFlexElt.style.gap = "4px";
```
Comment (peaBerberian) [inline inspector/src/modules/log_module.ts:130]: :O Didn't know about that one

**Pull request merged**